### PR TITLE
Formatted phase_angle tool as well as including error classes

### LINKED ISSUE #82: Tools phase angle needs type version for parameter
In tools, phase_angle needs "state" to be "quantum_state" and also needs a union type of either a numpy array or a quantumcircuit object.

### PR DESCRIPTION
**Pull request by brennanfreeze**
# Pull Request

## Description

Please include a summary of the changes and the motivation behind them. If applicable, link to any related issues.

## Type of Change

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Documentation
- [ ] Refactoring
- [ ] Other (please specify): 

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the documentation accordingly.
- [x] I ran `black` formatter on the codebase. (CI/CD will fail if neglected)
- [x] My code follows the project's coding style guidelines.

## Related Issues

- Closes #82 #97 
- Related to #

## Screenshots (if applicable)

If your change includes visual updates, please include screenshots to demonstrate the changes.

## Additional Notes

Add any additional notes or comments here. For example, describe any known issues or limitations.

### CODE REVIEW

File: src/tools/phase_angle.py [hunk 1]
```diff
@@ -1,35 +1,54 @@
...
-def phase_angle(state, show_bit=-1, round: int = 3, radian: bool = True):
-
-    size = int(log2(state.size))
-
-    state = convert_state(state)
     if round < 0:
-        exit(
-            f"Error: QuantumCircuit.tools.phaseAngle() -- round placement must be a value greater than 0."
-        )
+        RoundBelowZero(f"Cannot round to {round} needs to be 0 or greater")
```
Comment (parisosuch-dev) [inline src/tools/phase_angle.py:32]: correct me if im wrong but wouldn't you want to do `raise RoundBelowZero()`? Maybe I didn't know you can raise an error in python without explicitly using the keyword `raise`

File: src/tools/phase_angle.py [hunk 2]
```diff
@@ -1,35 +1,54 @@
...
-
-        if isinstance(show_bit, str):
-            phaseangle = int(show_bit, 2)
-        if 2**size <= phaseangle:
-            exit(
-                f"Error: QuantumCircuit.tools.phaseAngle() -- Called bit to find phase angle is not within range of possible values."
+    elif isinstance(show_bit, str):
+        phaseangle = int(show_bit, 2)
+        if 2**size <= phase_angle:
+            OutOfRangeError(
```
Comment (parisosuch-dev) [inline src/tools/phase_angle.py:40]: Same with this one as well.

File: src/tools/phase_angle.py [hunk 3]
```diff
@@ -1,35 +1,54 @@
...
+                f"Cannot show bit of value {show_bit} as it is out of range"
             )
+        phaseangle = mod(angle(state[phaseangle]), 2 * pi) * (180 / pi)
 
+    elif isinstance(show_bit, int):
+        phaseangle = show_bit
+        if 2**size <= phase_angle:
+            OutOfRangeError(
```
Comment (parisosuch-dev) [inline src/tools/phase_angle.py:48]: and this one

**Review by parisosuch-dev — CHANGES_REQUESTED** [review]
I think we are missing the raise of an error here

File: src/tools/phase_angle.py (hunk 3 above)
Comment (brennanfreeze) [reply src/tools/phase_angle.py]: Thanks for spotting this, updated this!

**Review by parisosuch-dev — APPROVED** [review]
sweet!

**Pull request merged**